src/Moq/Obsolete/MockFactory.cs

```diff
@@ -363,8 +363,18 @@ public virtual void VerifyAll()
 		}
 
 		/// <summary>
-		/// Invokes <paramref name="verifyAction"/> for each mock 
-		/// in <see cref="Mocks"/>, and accumulates the resulting 
+		/// Calls <see cref="Mock.VerifyNoOtherCalls"/> on all mocks created by this factory.
+		/// </summary>
+		/// <seealso cref="Mock.VerifyNoOtherCalls"/>
+		/// <exception cref="MockException">One or more mocks had setups that were not satisfied.</exception>
+		public virtual void VerifyNoOtherCalls()
+		{
+			VerifyMocks(Mock.VerifyNoOtherCalls);
```

Comment:
I suspect that this call won't do proper error aggregation. Let's say I'm adding another test for a repository with more than one mock:

```csharp
[Fact]
public void VerifyNoOtherCalls_should_correctly_aggregate_unmatched_calls_from_more_than_one_mock()
{
    var repository = new MockRepository(MockBehavior.Default);
    repository.Create<IFoo>().Object.Do();
    repository.Create<IBar>().Object.Redo();

    var ex = Record.Exception(() => repository.VerifyNoOtherCalls());

    Assert.Contains("The following invocations were not verified:", ex.Message);
    Assert.Contains("IFoo.Do()", ex.Message);
    Assert.Contains("IBar.Redo()", ex.Message);
}
```

This test will fail because the error message is this:

```
The following invocations were not verified:
IFoo.Do()
```

No mention of the `mock2.Redo()` call.

Could you please add a test such as the above and look into error aggregation for `VerifyNoOtherCalls`?

As an inspiration, here's how errors are aggregated for e.g. `VerifyAll`. Take this code:

```csharp
var repo = new MockRepository(MockBehavior.Default);
repo.Create<IFoo>().Setup(m => m.Do());
repo.Create<IBar>().Setup(m => m.Redo());

repo.VerifyAll();
```

This will produce an error message such as the following:

```
The following setups on mock 'Mock<IFoo:00000001>' were not matched:
IFoo m => m.Do()

The following setups on mock 'Mock<IBar:00000001>' were not matched:
IBar m => m.Redo()
```

Reply:
This threw up an interesting issue of multiple types of exception  needing to be grouped, so i added a new exception reason for this.